test/tests/selftest/pbs_default_timeout.py

```diff
@@ -36,32 +36,57 @@
 # trademark licensing policies.
 
 from tests.selftest import *
+from ptl.utils.plugins.ptl_test_runner import TimeOut
 
 
 class TestDefaultTimeout(TestSelf):
     """
     Test suite to verify increase default testcase timeout working properly.
     """
 
+    @staticmethod
+    def common_sleep_function(sleep_val):
+        time.sleep(sleep_val)
+
     def test_default_timeout(self):
         """
         Verify that test not timedout after 180 sec.
         i.e. after previous default timeout value
         """
-        time.sleep(300)
+        try:
+            self.common_sleep_function(MINIMUM_TESTCASE_TIMEOUT/2)
+        except TimeOut:
+            mssg = "Test Timeout before complete default-testcase-timeout"
+            raise TimeOut(mssg)
```

Comment:
don't raise TimeOut exception, that would have been the same thing as letting it timeout without catching it. we want to just let the user know that the test failed. So, maybe do "self.fail(\<error message>)" instead of raising TimeOut?

Reply:
Strange, because that is not what the instructions say: https://docs.python.org/2/library/unittest.html#skipping-tests-and-expected-failures
It says that it's more like a skip since it's an expected failure.

Reply:
If you mean this line: " In addition, it supports marking a test as an “expected failure,” a test that is broken and will fail, but shouldn’t be counted as a failure on a TestResult."

Then that refers to the decorator "@unittest.expectedFailure", not the unittest.fail() function. If you look at the logs that I posted, self.fail() basically raises an AssertionError

Reply:
I read it a bit too fast.  Thanks for pointing it out.  We can stop using self.assertTrue(False) now.  This will work for us.  Thanks for testing it out @agrawalravi90